index.js

```diff
@@ -8,6 +8,7 @@ var setSize = hasSet && setSizeDescriptor && typeof setSizeDescriptor.get === 'f
 var setForEach = hasSet && Set.prototype.forEach;
 var booleanValueOf = Boolean.prototype.valueOf;
 var objectToString = Object.prototype.toString;
+var getPrototype = Object.getPrototypeOf || function(o) { return o.__proto__; };
```

Comment:
What about the older engines where `__proto__` isn't even available?

Reply:
In that case, `__proto__` will be `undefined`, as will be the result of `getPrototype`. Consequently, `getTypeString` will return null (because the result of `getPrototype` is falsey). And the code will behave as it always has, *not* printing a type prefix.

Reply:
Wouldn't a `{ __proto__: 3 }` return `3` in an engine where it's not available?

Reply:
> Wouldn't a `{ __proto__: 3 }` return `3` in an engine where it's not available?

That will happen even in engines [that support `__proto__`](https://tc39.es/ecma262/#sec-__proto__-property-names-in-object-initializers).

Reply:
@ExE-Boss that spec text only treats it as a normal data property when it's not a string, or computed.

`({ __proto__: 3 }).__proto__ === Object.prototype` returns `true` in a modern engine, but in, say, IE 8, it returns `false`.

Reply:
Specifically, in IE 8, `({ __proto__: 3 }).__proto__` returns `3`.